packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -805,12 +818,21 @@ class _PageViewState extends State<PageView> {
 
     return NotificationListener<ScrollNotification>(
       onNotification: (ScrollNotification notification) {
-        if (notification.depth == 0 && widget.onPageChanged != null && notification is ScrollUpdateNotification) {
-          final PageMetrics metrics = notification.metrics as PageMetrics;
-          final int currentPage = metrics.page.round();
-          if (currentPage != _lastReportedPage) {
-            _lastReportedPage = currentPage;
-            widget.onPageChanged(currentPage);
+        if (notification.depth == 0) {
+          if (notification is ScrollUpdateNotification &&
+              widget.onPageChanged != null) {
+            final PageMetrics metrics = notification.metrics as PageMetrics;
+            final int currentPage = metrics.page.round();
+            if (currentPage != _lastReportedPage) {
+              _lastReportedPage = currentPage;
+              widget.onPageChanged(currentPage);
+            }
+          } else if (notification is ScrollCancelledNotification &&
```

Comment:
Do we really need to add a new Notification type? For this case, wouldn't you just get a `ScrollEndNotification` and from the PageMetrics included in there you could infer that the page hasn't changed by looking at the `page` property and comparing it to the value it had in the ScrollStartNotification?